src/net_processing.cpp

```diff
@@ -996,6 +1016,33 @@ static bool CanServeWitnesses(const Peer& peer)
     return peer.m_their_services & NODE_WITNESS;
 }
 
+void PeerManagerImpl::UpdateChainTipSet(std::set<const CBlockIndex*>& chain_tips, const CBlockIndex* potential_new_tip)
+{
+    AssertLockHeld(cs_main);
+    if (!potential_new_tip) return;
```

Comment:
What's the reason of allowing a nullptr here, but then terminating if it's passed?
Perhaps this check is better before the call, and here the parameter should be non-nullptr?

Reply:
Changed this to take a reference to CBlockIndex, since most of the call sites where guaranteed to pass a non-null index anyway.